test/controllers/orders_controller_test.rb

```diff
@@ -37,4 +36,122 @@
       expect(session[:cart_id]).must_be_nil
     end
   end
+
+  describe "purchase" do
+    let(:product) { products(:product_1) }
+    let(:product2) { products(:product_2) }
+    let(:item_params) { { item: { quantity: 2 } } }
```

Comment:
Wow that's a lot of curly braces. Would you mind walking me through this so I can understand how it works?